### DISCUSSION TITLE
Prisma upgrade 3.12 and up fails due to error finding "rhel-openssl-1.1.x"

### DISCUSSION THREAD

**Discussion by gabyrfp** [prisma/prisma#15965] [Q&A]
We have been trying to upgrade our Prisma version past 3.11, but are facing the following error when upgrading to 3.12 and above:

`Query engine library for current platform "rhel-openssl-1.1.x" could not be found.`

We are using Docker, ncc and prisma to package up our code. Checking the docker file at the `/opt/app/node_modules/.prisma/client` directory which is one of the directories that Prisma used to be looking for the binary, I can see that the binary is indeed there, yet it still cannot find it.  The full error message:

```

Error: Query engine library for current platform "rhel-openssl-1.1.x" could not be found.
You incorrectly pinned it to rhel-openssl-1.1.x

This probably happens, because you built Prisma Client on a different platform.
(Prisma Client looked in "/snapshot/build._build/packages/control-plane/dist/ncc/libquery_engine-rhel-openssl-1.1.x.so.node")

Searched Locations:

  /snapshot/build._build/packages/.prisma/client
  [LOCAL_PATH] /plz-out/tmp/_yarn#install._build/packages/control-plane/node_modules/.prisma/client
  /snapshot/build._build/packages/control-plane/dist
  /snapshot/build._build/packages/control-plane/dist/ncc
  /snapshot/build._build/packages/control-plane/dist/ncc
  /tmp/prisma-engines
  /snapshot/build._build/packages/control-plane/dist/ncc

You already added the platforms "darwin", "rhel-openssl-1.1.x" to the "generator" block
in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
but something went wrong. That's suboptimal.

Please create an issue at https://github.com/prisma/prisma/issues/new
    at LibraryEngine.getLibQueryEnginePath (/snapshot/build._build/packages/control-plane/dist/webpack:/@conveyorai/control-plane/node_modules/.prisma/client/runtime/index.js:44345:1)
    at LibraryEngine.loadEngine (/snapshot/build._build/packages/control-plane/dist/webpack:/@conveyorai/control-plane/node_modules/.prisma/client/runtime/index.js:44345:1)
    at LibraryEngine.instantiateLibrary (/snapshot/build._build/packages/control-plane/dist/webpack:/@conveyorai/control-plane/node_modules/.prisma/client/runtime/index.js:44345:1)
```

I can see that now it is looking for the files in a snapshot directory? Is there a way to extend these search paths so that it can search in the directory where it is already located perhaps?

Has anyone else experienced something similar?

**Comment by janpio** [comment]
`/snapshot/` usually indicates you are using the CLI built via `pkg` (which is common for other language Prisma Clients, like Python or Go).

What command are you running when you get this error message?
What does your project setup and Prisma schema look like? Did you indeed bin something via a `binaryTarget`?

**Reply by gabyrfp** [reply]
I am just running my application as a docker image when I see this error message. We use `ncc` and `pkg` to bundle up all of our typescript code into a single executable binary. We also run `yarn prisma generate`.

**Reply by gabyrfp** [reply]
```
generator client {
  provider        = "prisma-client-js"
  output          = "../node_modules/.prisma/client"
  previewFeatures = ["interactiveTransactions"]
  binaryTargets   = ["native", "rhel-openssl-1.1.x"]
}
```

**Reply by gabyrfp** [reply]
Is there a way to override the path at which prisma is searching for this binary?

**Reply by gabyrfp** [reply]
We found that it appears that Prisma changed the way it searches for the binaries https://github.com/prisma/prisma/blob/708091fd37f0358aa5617f10a69056bfcbc023a1/packages/engine-core/src/library/LibraryEngine.ts#L503-L510, so we are wondering if this can be configured

**Reply by janpio** [reply]
Prisma is probably not compatible with "We use ncc and pkg to bundle up all of our typescript code into a single executable binary." out of the box. If you have a minimal reproduction for this, please open a bug report and provide all the information the issue template asks for.

**Reply by pantharshit00** [reply]
Hello @gabyrfp,

Did you get a chance to open the bug report? Please link it here if you have already done that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
